src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/model/AddressDeleteReason.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.model
+
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import jakarta.validation.constraints.NotNull
+import uk.gov.justice.digital.hmpps.hmppsassessforearlyreleaseapi.model.enum.AddressDeleteReasonType
+import kotlin.reflect.KClass
+
+@ValidAddressDeleteReason
+@Schema(description = "Records an offender's non disclosable information")
+data class AddressDeleteReason(
+  @Schema(description = "The reason why address deleted", example = "NO_LONGER_WANTS_TO_BE_RELEASED_HERE")
+  @field:NotNull
+  val addressDeleteReasonType: AddressDeleteReasonType,
+
+  @Schema(description = "Other reason to delete address", example = "Give details of information regards delete address")
+  val addressDeleteOtherReason: String?,
+)
+
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [AddressDeleteReasonValidator::class])
+annotation class ValidAddressDeleteReason(
```

Comment:
In the past I would put annotations in there own package, something like model.valtiation.annotations,
Looks like we put it in with model kt file, which is fine until you use them in two dto's